keras_nlp/models/llama/llama_attention.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from keras_nlp.backend import keras
+from keras_nlp.backend import ops
+from keras_nlp.layers.modeling.rotary_embedding import RotaryEmbedding
+from keras_nlp.models.llama.dynamic_ntk_scaled_rotary_embedding import (
+    DynamicNTKScaledRotaryEmbedding,
+)
+from keras_nlp.utils.keras_utils import clone_initializer
+
+
+class LlamaAttention(keras.layers.Layer):
+    def __init__(
+        self,
+        num_heads,
+        hidden_dim,
```

Comment:
I don't think you need to pass `hidden_dim`, you just have it in `build()` right?

Also probably clearer to do `num_query_heads` and `num_key_value_heads`.